build/source/engine/snwCompact.f90

```diff
@@ -178,8 +178,8 @@ subroutine snwDensify(&
  endif
 
  ! check for low/high snow density
- if(any(mLayerVolFracIceNew(1:nSnow)*iden_ice < 50._dp) .or. &
-    any(mLayerVolFracIceNew(1:nSnow)*iden_ice > 900._dp))then
+ if(any(mLayerVolFracIceNew(1:nSnow)*iden_ice < 40._dp) .or. &
+    any(mLayerVolFracIceNew(1:nSnow)*iden_ice > 917._dp))then
```

Comment:
Avoid hard-coded numbers. Some more questions in this case:

`iden_ice` is already defined in `dshare/multiconst.f90` as `917.0_dp`, so the line 182 may read better as:

`    mLayerVolFracIceNew(1:nSnow)*iden_ice > iden_ice))then`
or 
`   mLayerVolFracIceNew(1:nSnow) > 1._dp))then`

For line 181, it is not clear how the `40.` relates to the `150.` for the minimum density that you use earlier for new snow. Underscoring the potential confusion of using hard-coded variables.